### PR TITLE
Make ConfigItem short_option optional (#4136)

### PR DESCRIPTION
**Pull request by lacostej**
#4136

Also cleaned up the specs

### CODE REVIEW

File: fastlane_core/spec/configuration_spec.rb [hunk 1]
```diff
@@ -85,113 +82,14 @@
...
-              foo: "",
-              bar: ""
-          }
-
-          expect do
-            FastlaneCore::Configuration.create(conflicting_options, values)
-          end.to raise_error "You can't use option '-b' along with '-f'"
-        end
-
-        it "raises an error for unresolved conflict between options" do
```
Comment (KrauseFx) [inline fastlane_core/spec/configuration_spec.rb]: Why did you remove those tests?

Comment (lacostej) [inline fastlane_core/spec/configuration_spec.rb]: Those tests are duplicated.

I submitted this PR as 2 commits: https://github.com/fastlane/fastlane/pull/4292/commits

Comment (lacostej) [inline fastlane_core/spec/configuration_spec.rb]: image

**Comment by mfurtak** [comment]
Looks good to me. Thanks! 👍
